From 4fea4a7c37b42cb41b3b7ad5a2d8dae2f27fe037 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Wed, 21 Mar 2012 15:28:17 +1000
Subject: [PATCH 1/6] Check ioctl return value

Really, this shouldn't ever fail but at least it shuts up some static
analysers.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/wcmUSB.c |   11 +++++++++--
 1 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/wcmUSB.c b/src/wcmUSB.c
index a2d30da..096943e 100644
--- a/src/wcmUSB.c
+++ b/src/wcmUSB.c
@@ -1563,11 +1563,18 @@ static void usbDispatchEvents(InputInfoPtr pInfo)
 	if (!ds->device_type && !dslast.proximity)
 	{
 		unsigned long keys[NBITS(KEY_MAX)] = { 0 };
+		int rc;
 
 		/* Retrieve the type by asking a resend from the kernel */
-		ioctl(common->fd, EVIOCGKEY(sizeof(keys)), keys);
+		rc = ioctl(common->fd, EVIOCGKEY(sizeof(keys)), keys);
+		if (rc == -1)
+		{
+			xf86Msg(X_ERROR, "%s: failed to retrieve key bits\n",
+					pInfo->name);
+			return;
+		}
 
-		for (i=0; i < ARRAY_SIZE(wcmTypeToKey); i++)
+		for (i = 0; i < ARRAY_SIZE(wcmTypeToKey); i++)
 		{
 			if (ISBITSET(keys, wcmTypeToKey[i].tool_key))
 			{
-- 
1.7.7.6

